tests/unit/amo/pages/TestLanguageTools.js

```diff
@@ -90,157 +81,129 @@ describe(__filename, () => {
     }),
   ];
 
-  function renderShallow({
-    i18n = fakeI18n(),
-    store = dispatchClientMetadata().store,
-    ...props
-  } = {}) {
-    return shallowUntilTarget(
-      <LanguageTools i18n={i18n} store={store} {...props} />,
-      LanguageToolsBase,
-    );
+  function render({ store = dispatchClientMetadata().store } = {}) {
+    return defaultRender({
+      history: createHistory({
+        initialEntries: [`/${lang}/firefox/language-tools/`],
+      }),
+      store,
+    });
   }
 
   it('renders LoadingText if language tools are not set', () => {
-    const root = renderShallow();
+    render();
 
-    expect(root.find(LoadingText)).not.toHaveLength(0);
+    expect(screen.queryAllByClassName('LoadingText').length).toBeGreaterThan(0);
   });
 
   it('renders LoadingText if language tools are empty', () => {
-    const { store } = dispatchClientMetadata();
+    const { store } = dispatchClientMetadata({ lang });
     store.dispatch(loadLanguageTools({ languageTools: [] }));
 
-    const root = renderShallow({ store });
+    render({ store });
 
-    expect(root.find(LoadingText)).not.toHaveLength(0);
+    expect(screen.queryAllByClassName('LoadingText').length).toBeGreaterThan(0);
   });
 
   it('renders language tools in your locale', () => {
     const { store } = dispatchClientMetadata({ lang: 'fr' });
     store.dispatch(loadLanguageTools({ languageTools }));
 
-    const root = renderShallow({ store });
+    render({ store, lang: 'fr' });
 
-    const dictionary = root.find(
-      `.LanguageTools-in-your-locale-list-item--${ADDON_TYPE_DICT}`,
-    );
-    const langPack = root.find(
-      `.LanguageTools-in-your-locale-list-item--${ADDON_TYPE_LANG}`,
+    expect(
+      screen.getByRole('heading', { name: 'Available for your locale' }),
+    ).toBeInTheDocument();
+
+    const dictionary = screen.getByClassName(
+      `LanguageTools-in-your-locale-list-item--${ADDON_TYPE_DICT}`,
     );
+    expect(
+      within(dictionary).getByText('le French Dictionary'),
+    ).toBeInTheDocument();
 
-    expect(root.find(LoadingText)).toHaveLength(0);
-    expect(root.find('.LanguageTools-in-your-locale')).toHaveLength(1);
-    expect(dictionary).toHaveLength(1);
-    expect(dictionary.find(Link)).toHaveProp(
-      'children',
-      'le French Dictionary',
+    const langPack = screen.getByClassName(
+      `LanguageTools-in-your-locale-list-item--${ADDON_TYPE_LANG}`,
     );
-    expect(langPack).toHaveLength(1);
-    expect(langPack.find(Link)).toHaveProp('children', 'French Language Pack');
+    expect(langPack).toBeInTheDocument();
+    expect(
+      within(langPack).getByText('French Language Pack'),
+    ).toBeInTheDocument();
   });
 
   it('omits "language tools in your locale" section if none available', () => {
     const { store } = dispatchClientMetadata({ lang: 'pt-BR' });
     store.dispatch(loadLanguageTools({ languageTools }));
 
-    const root = renderShallow({ store });
+    render({ store, lang: 'pt-BR' });
```

Comment:
As mentioned above, this `lang` won't do anything, so I'm not sure why this is even working. Maybe the `dispatchClientMetadata` is enough, and the test doesn't need the new `lang` in the `location`, but I would look into this a bit.